openhtf/conf.py

```diff
@@ -206,62 +206,16 @@ def __init__(self, flags, logger, lock, _functools, **kwargs):
     self._declarations = {}
 
     # Parse just the flags we care about, since this happens at import time.
-    sys.argv = self._ParseSingleFlag(sys.argv, 'config-file')
-    sys.argv = self._ParseSingleFlag(sys.argv, 'config-value')
+    self._flags, sys.argv = _ARG_PARSER.parse_known_args()
 
     # Populate flag_values from flags now.
     self._flag_values = {}
-    for keyval in flags['config-value'].value:
+    for keyval in self._flags.config_value:
       self._flag_values.setdefault(*keyval.split('=', 1))
 
     # Initialize self._loaded_values and load from --config-file if it's set.
     self.Reset()
 
-  def _ParseSingleFlag(self, argv, flag_name):
-    """Parse a single gflag from argv.
-
-    This duplicates some of the parsing done in gflags, but we do it here so we
-    can access certain flags at module import time, without forcing the user to
-    have to parse all commandline flags beforehand.  We also don't want to parse
-    all commandline flags as a side effect of loading this module, so we do this
-    bit of hackery to parse just the ones we care about.
-
-    Args:
-      argv: sys.argv to parse, new argv will be returned, like gflags.FLAGS().
-      flag_name: flag name to parse for, will raise if it's not registered.
-    """
-    if len(argv) < 2:
-      return argv
-
-    flag = self._flags.FlagDict()[flag_name]
-    indices_to_delete = set()
-    for idx, arg in enumerate(argv[1:], start=1):
-      if arg.startswith('-'):
-        arg = arg.lstrip('-')
-        if '=' in arg:
-          # '--config-file=foo.json' syntax, split on the first '='.
-          name, value = arg.split('=', 1)
-        elif idx + 1 >= len(argv):
-          # If no value is given, skip this token.  gflags actually raises in
-          # this case, but we'll just ignore it and let gflags do the raising
-          # when the arguments are parsed for real.
-          continue
-        else:
-          # '--config-file foo.json' syntax, grab the entire next arg.
-          name, value = arg, argv[idx + 1]
-        # If this flag matches the name we want, set the value.
-        if name == flag_name:
-          flag.Parse(value)
-          indices_to_delete.add(idx)
-          if '=' not in arg:
-            indices_to_delete.add(idx + 1)
-
-    # If we found at least one value, mark the flag as not using default.
-    flag.using_default_value = False
-
-    # Return a copy of argv, minus any args we parsed.
-    return [v for i, v in enumerate(argv) if i not in indices_to_delete]
-
```

Comment:
+1, really like that this goes away, it made me sad to write it in the first place, but gflags doesn't have a mechanism for doing this.